### PR TITLE
Create Openshift provider with Prometheus endpoint

### PR DESCRIPTION
**Pull request by zgalor**
Replaced `metrics: True` with default hawkular to generic endpoint terminology:
`monitoring: 'hawkular' / 'prometheus' / None`

endpoint params replaced accordingly:
`hawkular_hostname` => `monitoring_hostname`
`hawkular_port` => `monitoring_port`

Also added a creation example for prometheus in examples/add_openshift_provider.yml
and a test in tests/test_manageiq_provider.py

Temporary hack - No validation for prometheus endpoint authentication until the is supported by manageiq.

**Note**: this changes the module API and older playbooks with hawkular endpoint will not be supported from now on

### CODE REVIEW

**Comment by zgalor** [comment]
@cben @dkorn @moolitayer please review

File: examples/add_openshift_provider.yml
Comment (cben) [inline examples/add_openshift_provider.yml]: So this needs 4 parameters, and could be 2 or 3.
`metrics: true` and `metrics_type: 'prometheus'` are IMO redundant.  I propose combining to `metrics: 'prometheus'`.

In theory, both could be infered from the fact you provided `prometheus_hostname:` and `prometheus_port:`, but IIRC we added separate `metrics: true` boolean to be able to ~~specify it but omit host/port which had a sensible default(?)~~

@dkorn @enoodle what do you think?

Comment (zgalor) [reply examples/add_openshift_provider.yml]: So how about:
metrics : 'hawkular' / 'prometheus' / none (by default)
Then of course we could have `metrics_port, metrics_hostname` regardless of the venue.

But this is not backwards compatible with old playbooks. do we want it to be?

Comment (cben) [reply examples/add_openshift_provider.yml]: Actually I don't see any trace of hawkular host/port being optional.

**Review by cben — COMMENTED** [review]
Code LGTM. (aside from the interface metrics/metrics_type discussion)

File: examples/add_openshift_provider.yml
Comment (moolitayer) [reply examples/add_openshift_provider.yml]: I like best the option of having `hawkular_hostname`, `hawkular_port` OR `prometheus_hostname`, `prometheus_port`. Bad combinations should fail.
I would not touch `metrics: true` in this PR. 
I don't think we should be concerned with backward compatibility at this stage.

File: library/manageiq_provider.py
Comment (zgalor) [inline library/manageiq_provider.py:565]: I am not so happy with writing this validation twice - once for prometheus and once for hawkular, but I couldn't find a way in `required_if` to state "not None". Suggestions are welcome...

**Comment by zgalor** [comment]
After a consult with @moolitayer I decided to rename the endpoint parameters to `monitoring`:
```
monitoring: prometheus / hawkular / None
monitoring_hostname
monitoring_port
```

This aligns with the new naming convention in manageiq, and more generic for future usage of logging etc.

@cben @yaacov @dkorn @moolitayer  please review

File: examples/add_openshift_provider.yml
Comment (yaacov) [inline examples/add_openshift_provider.yml:20]: I think it's nicer if the `provider_verify_ssl` will be together with the `provider_api_hostname` and `api_port`

File: library/manageiq_provider.py
Comment (cben) [reply library/manageiq_provider.py:565]: I don't think there is.
https://github.com/ansible/ansible/blob/b69c7f50d02dac3e24bf1927ebdc933475da93b7/lib/ansible/module_utils/basic.py#L1635 checks `== val`.
Are you unhappy with textual duplication (LGTM) or the fact we'll have to remember to extend it if we add another?

required_if is not magical.
If you want you can check the condition yourself and exit with error (like done below for  `['miq_url', 'miq_username', 'miq_password']`)...

**Comment by cben** [comment]
If "monitoring" term can include logging, will we at some point need 2 monitoring endpoints of different kinds?

**Comment by zgalor** [comment]
@cben Probably not. More likely that each will reside in a different manager, @moolitayer keep me honest here.

File: examples/add_openshift_provider.yml
Comment (zgalor) [reply examples/add_openshift_provider.yml:20]: I agree, but this is only an example, and also irrelevant to this PR

File: library/manageiq_provider.py
Comment (zgalor) [reply library/manageiq_provider.py:565]: I am unhappy with both, but then again it might be clearer to leave it like this. I the list or choices grows longer, we should refactor.

File: library/manageiq_provider.py
Comment (cben) [inline library/manageiq_provider.py:556]: Could add `type='str'`, also missing on many others. Not necessarily in this PR.

File: library/manageiq_provider.py
Comment (cben) [inline library/manageiq_provider.py]: I'd pass these by name
```
generate_auth_key_config(
    role=monitoring,
    authtype=monitoring,
    ...
```

File: tests/test_manageiq_provider.py
Comment (cben) [inline tests/test_manageiq_provider.py]: consider s/openshift_without_metrics/openshift_without_monitoring/g.

more importantly, there is `miq_ansible_module.params = {'metrics': True}` on L274

**Review by cben — COMMENTED** [review]
Looks good.

File: library/manageiq_provider.py
Comment (zgalor) [reply library/manageiq_provider.py:556]: since we have "choices", this seems obsolete

File: library/manageiq_provider.py
Comment (zgalor) [reply library/manageiq_provider.py]: Very well.
So I guess this should be done for other calls, right?

File: tests/test_manageiq_provider.py
Comment (zgalor) [reply tests/test_manageiq_provider.py]: nice catch. thanks

File: library/manageiq_provider.py
Comment (cben) [reply library/manageiq_provider.py]: Your call ;-).
Personally I find I can't read the current `generate_auth_key_config()` call without going to see its definition.  I suppose if you change this, should also change the 2 other endpoint-generating calls.

File: tests/test_manageiq_provider.py
Comment (zgalor) [reply tests/test_manageiq_provider.py]: I wonder how the test passed with the `metrics: true` param. I will fix the test

**Comment by zgalor** [comment]
I think this is ready now.
@cben @moolitayer please take a final look

**Review by cben — APPROVED** [review]

**Comment by cben** [comment]
@dkorn I expect you have new priorities :) I'm starting to merge without reviews from you; let us know if you want to stay in the loop.

**Comment by cben** [comment]
@zgalor please edit PR description according to current interface.

**Comment by zgalor** [comment]
done.

**Comment by moolitayer** [comment]
:tada:

**Pull request merged**
